### PR TITLE
Support for Nextstrain CLI's new means of authentication with IdPs

### PR DESCRIPTION
**Pull request by tsibley**
Nextstrain CLI will start using OIDC/OAuth2's authorization code flow to interact with not just AWS Cognito but other IdPs as well (i.e. as used in other deployments of nextstrain.org).

To support this without hardcoding or onerous user-side configuration, Nextstrain CLI will start using the standard OIDC configuration endpoint, /.well-known/openid-configuration, to auto-discover necessary configuration about both the IdP to talk to and the client it should be.

Related-to: <https://github.com/nextstrain/private/issues/94>

Related CLI PR: TKTK

## Checklist

- [x] Checks pass
- [ ] Apply Terraform changes to production environment in advance of deploy and update config.json

### CODE REVIEW

File: env/production/config.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "OIDC_IDP_URL": "https://cognito-idp.us-east-1.amazonaws.com/us-east-1_Cg5rcTged",
   "OAUTH2_CLIENT_ID": "rki99ml8g2jb9sm1qcq9oi5n",
   "OAUTH2_CLI_CLIENT_ID": "2vmc93kj4fiul8uv40uqge93m5",
+  "OAUTH2_CLI_CLIENT_REDIRECT_URIS": [],
```
Comment (tsibley) [inline env/production/config.json:6]: This should be filled out before deploy, after applying Terraform updates to production.

**Comment by tsibley** [comment]
This could get merged and deployed without affecting current Nextstrain CLI clients. The main changes are a new nextstrain.org server endpoint (used by work-in-progress Nextstrain CLI changes TKTK) and additional allowed auth flows for the CLI's client registration with Cognito.

**Review by victorlin — APPROVED** [review]

**Comment by tsibley** [comment]
@victorlin @jameshadfield Thoughts on if I should just deploy https://github.com/nextstrain/nextstrain.org/pull/757/commits/1019326ebc378648e51ee54cf1c5024ab7460b85 with this branch?

**Comment by tsibley** [comment]
(I'm inclined to just deploy it since it's read-only and public data.)

**Comment by victorlin** [comment]
Seems fine to me, it'll make the dev process easier.

**Pull request merged**